### DISCUSSION TITLE
How to get a dependency graph for which packages are including other packages (godepgraph)

### DISCUSSION THREAD

**Discussion by rcoreilly** [cogentcore/core#1272] [Tips and Tricks] [answered]
This is not exactly Cogent Core specific but I keep having to do this to figure out why `core/core` is getting included in the `goal` cmd, which then causes a GUI window to open.  Maybe we could provide a better GUI version of this functionality at some point?

* tool: https://github.com/kisielk/godepgraph

Example usage:
```sh
godepgraph -s -p github.com,gonum.org,golang.org . | dot -Tpng -o depgraph.png 
godepgraph -s -p github.com,gonum.org,golang.org . > depgraph
```

the .png is sufficiently complex that it is hard to use -- the depgraph is better but you have to pull it up in an editor and search -- that is where a nice tabular GUI might be handy.

**Comment by rcoreilly** [comment, marked as answer]
This is already answered above :)

**Comment by kkoreilly** [comment]
Yep, a GUI for this as part of a future Cogent Developer Tools app for Go alongside binary size analysis, profiling, and documentation would definitely be good.

**Comment by rcoreilly** [comment]
I guess we need a different mechanism (in our new wiki-like docs presumably) for random bits of info like this?  Once it is closed, it is much less discoverable..

**Reply by kkoreilly** [reply]
Random bits of info like this that have no relevance to Cogent Core don't really belong anywhere; once we make Cogent Developer Tools, this will be superseded anyway.
